lib/mongo/crypt/binding.rb

```diff
@@ -130,31 +182,71 @@ class Binding
         :debug,   4,
       ]
 
-      # Mongocrypt log function signature. Takes a log level, a log message as a string,
-      # an integer representing the length of the message, and a pointer to a context provided
-      # by the caller (can be set to nil).
+      # A callback to the mongocrypt log function
+      # Set a custom log callback with the mongocrypt_setopt_log_handler method
+      #
+      # @param [ Symbol ] level The log level; possible values defined by the
+      #   log_level enum
+      # @param [ String ] message The log message
+      # @param [ Integer ] len The length of the message param, or -1 if the
+      #   string is null terminated
+      # @param [ FFI::Pointer | nil ] ctx An optional pointer to a context
+      #   object when this callback was set
+      #
+      # @return [ nil ] Always nil.
       callback :mongocrypt_log_fn_t, [:log_level, :string, :int, :pointer], :void
 
-      # Sets a method to be called on every log message. Takes a pointer to a mongocrypt_t object,
-      # a mongocrypt_log_fn_t callback, and a pointer to a log_ctx. Returns a boolean indicating
-      # success of the operation.
-      attach_function :mongocrypt_setopt_log_handler, [:pointer, :mongocrypt_log_fn_t, :pointer], :bool
+      # Creates a new mongocrypt_t object
+      #
+      # @return [ FFI::Pointer ] A pointer to a new mongocrypt_t object
+      attach_function :mongocrypt_new, [], :pointer
+
+      # Set the handler on the mongocrypt_t object to be called every time
+      #   libmongocrypt logs a message
+      #
+      # @param [ FFI::Pointer ] crypt A pointer to a mongocrypt_t object
+      # @param [ Method ] log_fn A logging callback method
+      # @param [ FFI::Pointer | nil ] log_ctx An optional pointer to a context
+      #   to be passed into the log callback on every invocation.
+      #
+      # @return [ Boolean ] Whether setting the callback was successful
+      attach_function(
+        :mongocrypt_setopt_log_handler,
+        [:pointer, :mongocrypt_log_fn_t, :pointer],
+        :bool
+      )
 
+      # Set the logger callback function on the Mongo::Crypt::Handle object
+      #
+      # @param [ Mongo::Crypt::Handle ] handle
+      # @param [ Method ] log_callback
+      #
+      # @raise [ Mongo::CryptError ] If the callback is not set successfully
       def setopt_log_handler(handle, log_callback)
```

Comment:
This seems to be missing `self.` so it's not invoked anywhere?

Reply:
Or only on your local test runs with logging on.

Reply:
Yeah you're right only the local tests have logging. Adding `self`.